pkg/k8s/deployment.go

```diff
@@ -0,0 +1,207 @@
+package k8s
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/tigera/operator/pkg/render/common/secret"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Deployment struct {
+	dep appsv1.Deployment
+
+	// Not using a map because order can change and there's not so many containers that it would cause a performance issue.
+	ctrs []*Container
+}
+
+func NewDeployment(dep appsv1.Deployment) *Deployment {
+	return &Deployment{dep: dep}
+}
+
+func (d *Deployment) Deployment() *appsv1.Deployment {
+	return &d.dep
+}
+
+func (d *Deployment) SetPullSecrets(pullSecrets []*corev1.Secret) {
+	d.dep.Spec.Template.Spec.ImagePullSecrets = secret.GetReferenceList(pullSecrets)
+}
+
+func (d *Deployment) Services() []*corev1.Service {
+	var srvs []*corev1.Service
+	for _, ctr := range d.ctrs {
+		if ctr.srvc != nil {
+			ctr.srvc.SetNamespace(d.dep.ObjectMeta.Namespace)
+			srvs = append(srvs, ctr.srvc)
+		}
+	}
+	return srvs
+}
+
+func (d *Deployment) ApplyImageSet(installation *operatorv1.InstallationSpec, imageSet *operatorv1.ImageSet) error {
+	for _, ctr := range d.ctrs {
+		if err := ctr.ApplyImageSet(installation, imageSet); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (d *Deployment) MountedSecrets() []*corev1.Secret {
+	var scrts []*corev1.Secret
+	for _, ctr := range d.ctrs {
+		for _, scrt := range ctr.mountedSecrets {
+			scrt.SetNamespace(d.dep.ObjectMeta.Namespace)
+			scrts = append(scrts, scrt)
+		}
+	}
+
+	return scrts
+}
+
+func (d *Deployment) MountedConfigMaps() []*corev1.ConfigMap {
+	var cms []*corev1.ConfigMap
+	for _, ctr := range d.ctrs {
+		for _, cm := range ctr.mountedConfigMaps {
+			cm.SetNamespace(d.dep.ObjectMeta.Namespace)
+			cms = append(cms, cm)
+		}
+	}
+
+	return cms
+}
+
+func (d *Deployment) Key() string {
+	key := d.dep.ObjectMeta.Name
```

Comment:
ISTR there is already a library function that does this?

https://pkg.go.dev/k8s.io/client-go/tools/cache#MetaNamespaceKeyFunc

Not really a big deal though.